src/Protocol/MojangAPI.h

```diff
@@ -83,6 +79,34 @@ class cMojangAPI
 	/** Sets the m_RankMgr that is used for name-uuid notifications. Accepts nullptr to remove the binding. */
 	void SetRankManager(cRankManager * a_RankManager) { m_RankMgr = a_RankManager; }
 
+	class cCallbacks : public cUrlClient::cCallbacks
+	{
+	public:
+
+		explicit cCallbacks(std::shared_ptr<cEvent> a_Event, AString & a_ResponseBody) : m_Event(std::move(a_Event)), m_ResponseBody(a_ResponseBody) {}
+
+		void OnBodyFinished() override
+		{
+			m_Event->Set();
+		}
+
+		void OnError(const AString & a_ErrorMsg) override
+		{
+			LOGERROR("%s %d: HTTP Error: %s", __FILE__, __LINE__, a_ErrorMsg.c_str());
+			m_Event->Set();
+		}
+
+		void OnBodyData(const void * a_Data, size_t a_Size) override
+		{
+			m_ResponseBody.append(static_cast<const char *>(a_Data), a_Size);
+		}
+
+		std::shared_ptr<cEvent> m_Event;
+
+		/** The accumulator for the partial body data, so that OnBodyFinished() can send the entire thing at once. */
+		AString & m_ResponseBody;
+	};
+
```

Comment:
This could probably be pulled into the cpp file into an anonymous namespace.

Reply:
I'd like to have access to this in other classes. Maybe add this to `cUrlClient` as default or basic

Reply:
That's actually a good idea, move this completely to `cUrlClient` as `BlockingGet()` and `BlockingPost()` etc. methods. Just don't export them to Lua! :D